src/components/typography/typography.tsx

```diff
@@ -0,0 +1,36 @@
+import { useTheme } from '@react-navigation/native';
+import React from 'react';
+import { Text, TextProps } from 'react-native';
+
+import { styles } from './typography.styles';
+
+interface TypographyProps extends TextProps {
+  variant?: 'h1' | 'h2' | 'h3' | 'body' | 'caption';
+  style?: object;
```

Comment:
This is not expected.

By exposing `style` as props, the consumer can override it. This is seems like it would open opportunity for callers to customize the style which would lead to inconsistencies.